casadm/cas_lib.c

```diff
@@ -565,7 +604,68 @@ int get_dev_path(const char* disk, char* buf, size_t num)
 	return err;
 }
 
-int get_core_info(int fd, int cache_id, int core_id, struct kcas_core_info *info)
+/* Indicate whether given path should be passed without check */
+static bool is_dev_link_whitelisted(const char* path)
+{
+	regex_t regex;
+	int result;
+	static const char* const whitelist[] = {"/dev/cas[0-9]\\+-[0-9]\\+$"};
+	static const unsigned count = ARRAY_SIZE(whitelist);
+	size_t i;
+
+	for (i = 0; i < count; i++) {
+		result = regcomp(&regex, whitelist[i], REG_NOSUB);
+		if (result)
+			return FAILURE;
+
+		result = regexec(&regex, path, 0, NULL, 0);
+		regfree(&regex);
+		if (!result) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static int _is_by_id_path(const char* dev_path)
+{
+	static const char dev_by_id_dir[] = "/dev/disk/by-id";
+
+	return (!strncmp(dev_path, dev_by_id_dir,
+			 strnlen_s(dev_by_id_dir, sizeof(dev_by_id_dir))));
+}
+
+int set_device_path(char *dest_path, size_t dest_len, const char *src_path, size_t src_len)
+{
+	char abs_dev_path[MAX_STR_LEN] = {};
```

Comment:
I think initializing the path (= {}) here is not necessary

Reply:
@mmichal10 - we analyzed this particular case - zeroing this array wouldn't affect security, but can affect performance.